rtv.go

```diff
@@ -40,46 +43,56 @@ func StopCron() {
 // rtvPoll attempts to re-populate the RTVCache. If this is the very first time,
 // and there are any errors, this will die fatally.
 func rtvPoll() {
+	// Make a copy for transactional state.
+	newCache := *RTVCache
+
+	// Decide to die if this is the very first time initializing the value.
+	shouldDie := RTVCache.RTV == ""
+	maybeDie := func(err interface{}) {
+		if shouldDie {
+			die("%+v", err)
+		}
+		log.Print(err)
+	}
+
 	// Fetch the runtime version number
-	getRtv(&RTVCache.RTV, rtvHost+"/v0/version.txt")
+	// TODO(angielin): This is a temporary endpoint. Migrate to metadata
+	// endpoint when ready.
+	var err error
+	if newCache.RTV, err = getRTVBody(rtvHost + "/v0/version.txt"); err != nil {
+		// If there is a problem getting the RTV value, there is no need to
+		// get the CSS
+		maybeDie(err)
+		return
+	}
 	// Pad to width of 15
-	RTVCache.RTV = fmt.Sprintf("%015s", RTVCache.RTV)
+	newCache.RTV = fmt.Sprintf("%015s", newCache.RTV)
 
 	// Fetch the CSS payload
-	getRtv(&RTVCache.CSS, rtvHost+"/rtv/"+RTVCache.RTV+"/v0.css")
+	if newCache.CSS, err = getRTVBody(rtvHost + "/rtv/" + newCache.RTV + "/v0.css"); err != nil {
+		// If there was a problem getting CSS, abort and don't write new cache value.
+		maybeDie(err)
+		return
+	}
+	RTVCache = &newCache
 }
 
-// get retrieves the body contents of the given url, populating the value into
-// the given string pointer, fatally dying if there are any errors if this is the first time.
-func getRtv(s *string, url string) {
+// getRTVBody returns the body contents of the given url, or an error if there was problem.
+func getRTVBody(url string) (string, error) {
 	log.Printf("Fetching URL: %q\n", url)
-	resp, err := http.Get(url)
-	defer resp.Body.Close()
-	// Decide to die if this is the very first time initializing the value.
-	shouldDie := *s == "" || s == nil
-
+	resp, err := rtvClient.Get(url)
 	if err != nil {
-		if shouldDie {
-			die("%+v", err)
-		}
-		log.Printf("Error getting %s: %+v", url, err)
-		return
+		return "", err
```

Comment:
You can use `errors.Wrapf(err, "fetching %s", url)` to add context (ditto line 95).